test/test_nn.py

```diff
@@ -1600,6 +1600,16 @@ def test_Sequential_append(self):
         self.assertEqual(n2, nn.Sequential(l1, l2, l3, l4))
         self.assertEqual(nn.Sequential(l1).append(l2).append(l4), nn.Sequential(l1, l2, l4))
 
+    def test_Sequential_pop(self):
+        l1 = nn.Linear(1, 2)
+        l2 = nn.Linear(2, 3)
+        l3 = nn.Linear(3, 4)
+        l4 = nn.Linear(4, 5)
+        n1 = nn.Sequential(l1, l2, l3, l4)
+        self.assertEqual(l4, n1.pop(3))
+        n2 = nn.Sequential(l1, l2, l3)
+        self.assertEqual(n1, n2)
```

Comment:
Should we also check if the index of the Sequential is updated? (since in current state, the test passes even if we don't reorder) @albanD @khushi-411 

Thanks!

Reply:
Looks like the both ignored me when I asked for the merge :P So let's add this here and trigger the merge again!

Reply:
Yep, mandatory case to be added to avoid confusion. Thanks!

Added the test.